src/pages/TaskList/TaskList.js

```diff
@@ -1,88 +1,102 @@
-import './taskList.scss';
-import {Link} from 'react-router-dom';
-import {Tabs, Tab} from '../../components/Tabs/index';
-import {getTasks, deleteTask, updateTask} from '../../services/tasks';
-import { days } from '../../consts';
-
-export class TaskList extends React.Component {
-    constructor(props) {
-        super(props);
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
 
-        this.state = {
-            tasksInWeek: []
-        };
-    }
+import './taskList.scss';
+import { Tabs, Tab } from '../../components/Tabs/index';
+import { deleteTask, updateTask, getTasks as getTasksRequest } from '../../services/tasks';
+import { days } from '../../constants/consts';
+import { getTasks } from '../../store';
 
 
-    date = new Date().getDay();
+export class TaskListContainer extends React.Component {
+  date = new Date().getDay();
 
-    createTask = (day) => {
-        this.props.history.push(`tasks/newtask?day=${day}`);
-    };
+  createTask = (day) => {
+    this.props.history.push(`tasks/newtask?day=${day}`);
+  };
 
 
     updateTaskList = () => {
-        getTasks()
-          .then(tasksInWeek => this.setState({tasksInWeek}));
+        getTasksRequest()
+          .then(tasksInWeek => this.props.getTasks(tasksInWeek));
     };
 
     changeTask = (task) => {
-        let tasksInWeek = [...this.state.tasksInWeek];
+        const tasksInWeek = [...this.props.tasksInWeek];
+        console.log(tasksInWeek);
         updateTask(task)
-          .then(() => this.setState({ tasksInWeek }));
+          .then(() => this.setState({tasksInWeek}));
+    };
+
+
+    changeTaskState = (task, state = false) => {
+        task.done = state;
+        console.log(task.done);
+        this.changeTask(task);
+        console.log(task);
     };
 
+
     handleDeleteTask = (id) => {
         deleteTask(id)
           .then(() => this.updateTaskList());
```

Comment:
you shouldn't fetch all tasks again from server, remove task from array in redux store